File: posttest2/posttest2.js
const readline = require("readline-sync")

class UC {

    #id
    #namaPaket
    #jumlahUC
    #harga

    constructor(id, namaPaket, jumlahUC, harga) {
        this.#id = id
        this.#namaPaket = namaPaket
        this.#jumlahUC = jumlahUC
        this.#harga = harga
    }

    getId() {
        return this.#id
    }

    getNamaPaket() {
        return this.#namaPaket
    }

    getJumlahUC() {
        return this.#jumlahUC
    }

    getHarga() {
        return this.#harga
    }

    setNamaPaket(nama) {
        this.#namaPaket = nama
    }

    setJumlahUC(jumlah) {
        this.#jumlahUC = jumlah
    }

    setHarga(harga) {
        this.#harga = harga
    }
}

let dataUC = []

function tambahData() {

    console.log("\nTambah Paket UC")

    let id = readline.question("ID Paket: ")
    let nama = readline.question("Nama Paket: ")
    let jumlah = readline.questionInt("Jumlah UC: ")
    let harga = readline.questionInt("Harga: ")

    let paket = new UC(id, nama, jumlah, harga)

    dataUC.push(paket)

    console.log("Data berhasil ditambahkan")
}

function lihatData() {

    console.log("\nDaftar Paket UC")

    if (dataUC.length === 0) {
        console.log("Data masih kosong")
        return
    }

    for (let i = 0; i < dataUC.length; i++) {

        console.log(
            "ID:", dataUC[i].getId(),
            "| Nama:", dataUC[i].getNamaPaket(),
            "| UC:", dataUC[i].getJumlahUC(),
            "| Harga:", dataUC[i].getHarga()
        )
    }
}

function updateData() {

    let id = readline.question("Masukkan ID paket yang ingin diupdate: ")

    for (let i = 0; i < dataUC.length; i++) {

        if (dataUC[i].getId() === id) {

            let nama = readline.question("Nama paket baru: ")
            let jumlah = readline.questionInt("Jumlah UC baru: ")
            let harga = readline.questionInt("Harga baru: ")

            dataUC[i].setNamaPaket(nama)
            dataUC[i].setJumlahUC(jumlah)
            dataUC[i].setHarga(harga)

            console.log("Data berhasil diupdate")
            return
        }
    }

    console.log("Data tidak ditemukan")
}

function hapusData() {

    let id = readline.question("Masukkan ID paket yang ingin dihapus: ")

    for (let i = 0; i < dataUC.length; i++) {

        if (dataUC[i].getId() === id) {

            dataUC.splice(i, 1)

            console.log("Data berhasil dihapus")
            return
        }
    }

    console.log("Data tidak ditemukan")
}

function menu() {

    while (true) {

        console.log("\nSistem Penjualan UC PUBG Mobile")
        console.log("1. Tambah Paket UC")
        console.log("2. Lihat Paket UC")
        console.log("3. Update Paket UC")
        console.log("4. Hapus Paket UC")
        console.log("5. Exit")

        let pilih = readline.questionInt("Pilih menu: ")

        switch (pilih) {

            case 1:
                tambahData()
                break

            case 2:
                lihatData()
                break

            case 3:
                updateData()
                break

            case 4:
                hapusData()
                break

            case 5:
                console.log("Program selesai")
                return

            default:
                console.log("Menu tidak tersedia")
        }
    }
}

menu()

File: posttest2/posttest2.java
import java.util.ArrayList;
import java.util.Scanner;

class UC {
    private String id;
    private String namaPaket;
    private int jumlahUC;
    private int harga;

    public UC(String id, String namaPaket, int jumlahUC, int harga) {
        this.id = id;
        this.namaPaket = namaPaket;
        this.jumlahUC = jumlahUC;
        this.harga = harga;
    }

    public String getId() {
        return id;
    }

    public String getNamaPaket() {
        return namaPaket;
    }

    public int getJumlahUC() {
        return jumlahUC;
    }

    public int getHarga() {
        return harga;
    }

    public void setNamaPaket(String namaPaket) {
        this.namaPaket = namaPaket;
    }

    public void setJumlahUC(int jumlahUC) {
        this.jumlahUC = jumlahUC;
    }

    public void setHarga(int harga) {
        this.harga = harga;
    }
}

public class Main {

    static ArrayList<UC> dataUC = new ArrayList<>();
    static Scanner input = new Scanner(System.in);

    static void tambahData() {
        System.out.println("\nTambah Paket UC");

        System.out.print("ID Paket: ");
        String id = input.nextLine();

        System.out.print("Nama Paket: ");
        String nama = input.nextLine();

        System.out.print("Jumlah UC: ");
        int jumlah = input.nextInt();

        System.out.print("Harga: ");
        int harga = input.nextInt();
        input.nextLine(); // buang newline

        UC paket = new UC(id, nama, jumlah, harga);
        dataUC.add(paket);

        System.out.println("Data berhasil ditambahkan");
    }

    static void lihatData() {
        System.out.println("\nDaftar Paket UC");

        if (dataUC.isEmpty()) {
            System.out.println("Data masih kosong");
            return;
        }

        for (UC uc : dataUC) {
            System.out.println(
                "ID: " + uc.getId() +
                " | Nama: " + uc.getNamaPaket() +
                " | UC: " + uc.getJumlahUC() +
                " | Harga: " + uc.getHarga()
            );
        }
    }

    static void updateData() {
        System.out.print("Masukkan ID paket yang ingin diupdate: ");
        String id = input.nextLine();

        for (UC uc : dataUC) {
            if (uc.getId().equals(id)) {

                System.out.print("Nama paket baru: ");
                String nama = input.nextLine();

                System.out.print("Jumlah UC baru: ");
                int jumlah = input.nextInt();

                System.out.print("Harga baru: ");
                int harga = input.nextInt();
                input.nextLine();

                uc.setNamaPaket(nama);
                uc.setJumlahUC(jumlah);
                uc.setHarga(harga);

                System.out.println("Data berhasil diupdate");
                return;
            }
        }

        System.out.println("Data tidak ditemukan");
    }

    static void hapusData() {
        System.out.print("Masukkan ID paket yang ingin dihapus: ");
        String id = input.nextLine();

        for (int i = 0; i < dataUC.size(); i++) {
            if (dataUC.get(i).getId().equals(id)) {
                dataUC.remove(i);
                System.out.println("Data berhasil dihapus");
                return;
            }
        }

        System.out.println("Data tidak ditemukan");
    }

    static void menu() {
        while (true) {
            System.out.println("\nSistem Penjualan UC PUBG Mobile");
            System.out.println("1. Tambah Paket UC");
            System.out.println("2. Lihat Paket UC");
            System.out.println("3. Update Paket UC");
            System.out.println("4. Hapus Paket UC");
            System.out.println("5. Exit");

            System.out.print("Pilih menu: ");
            int pilih = input.nextInt();
            input.nextLine();

            switch (pilih) {
                case 1:
                    tambahData();
                    break;
                case 2:
                    lihatData();
                    break;
                case 3:
                    updateData();
                    break;
                case 4:
                    hapusData();
                    break;
                case 5:
                    System.out.println("Program selesai");
                    return;
                default:
                    System.out.println("Menu tidak tersedia");
            }
        }
    }

    public static void main(String[] args) {
        menu();
    }
}
